bootstrap/api/v1beta1/rke2config_types.go

```diff
@@ -146,12 +146,6 @@ type RKE2AgentConfig struct {
 	//+optional
 	LoadBalancerPort int `json:"loadBalancerPort,omitempty"`
 
-	// Version specifies the rke2 version.
-	// This field will be deprecated in newer versions of the API and RKE2ControlPlaneSpec.Version will be used instead.
-	// +kubebuilder:validation:Pattern="v(\\d\\.\\d{2}\\.\\d)\\+rke2r\\d"
-	//+optional
-	Version string `json:"version,omitempty"`
```

Comment:
We can leave the conversion of version for the RKE2ControlPlane but removing `Version` from RKE2Config for workers is a breaking change because version is managed from MachineDeployment object. I know the API is in alpha but we have users running it in production.

Reply:
In previous versions of CAPRKE2, we took the version from the `RKE2Config.AgentConfig.Version` to pass to the installation script https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/v0.2.7/bootstrap/internal/controllers/rke2config_controller.go#L745 for worker machines. MachineDeployment version field is placeholder field in all previous versions, which means that not everyone had this field configured properly(or configured at all, that's also a valid option). By removing `RKE2Config.AgentConfig.Version` we introduce a breaking change where users will lose data about current version of their worker machines.

Reply:
For those users who don’t have the MachineDeployment `spec.version` field configured properly or at all it indeed will be a breaking change. Under CAPI semantics this is invalid, as CP versions and worker versions supposed to be independent from each other. The problem is however in case when the version was configured properly, but is different from the CP version. These users as per #315 actually don’t have the expected version on the new worker nodes provisioned after the diff.

With this change we can be sure that all existing and new users of `v1beta1` do have declared MD/CP version matching the components version provisioned for the node.